src/pages/studyView/charts/barChart/BarChartToolTip.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from "react";
+import * as _ from 'lodash';
+import {computed} from "mobx";
+import { IAlterationData, IAlterationCountMap } from "pages/resultsView/cancerSummary/CancerSummaryContent";
+import { Popover } from "react-bootstrap";
+import * as ReactDOM from "react-dom";
+import classnames from "classnames";
+
+export type ToolTipRange = {
+    start: number | undefined,
+    end: number | undefined,
+    special: string | undefined
+};
+
+export type BarChartToolTipProps = {
+    mousePosition: { x: number, y: number },
+    totalBars: number,
+    currentBarIndex: number,
+    range: ToolTipRange | null,
+    sampleCount: number,
+    windowWidth: number,
+};
+
+const OFFSET = 17;
+const WIDTH = 150;
+
+export default class BarChartToolTip extends React.Component<BarChartToolTipProps, {}> {
+
+    /**
+     * When the active bar is past the middle of the plot, render on the left side of the bar
+     */
+    @computed
+    get placement(): "left" | "right" {
+        return (this.props.totalBars < this.props.currentBarIndex * 2) ? "left" : "right";
+    }
+
+    @computed
+    get positionLeft(): number {
+        return this.props.mousePosition.x + (this.placement == "left" ? (-OFFSET - WIDTH) : OFFSET); 
+    }
+
+    @computed
+    get transform(): string | undefined {
+        return (this.placement === "left" ? "translate(-100%,0%)" : undefined)
+    }
+
+    translateSpecialText(text: string | undefined): string {
+        if (!text) {
+            return "";
+        }
+
+        if (text === ">=") {
+            return "≥";
+        }
+
+        if (text === "<=") {
+            return "≤";
+        }
+
+        return text;
+    }
+
+
+    @computed
+    get rangeText(): string {
```

Comment:
The start and end need  to be converted using `toLocaleString`. Also the format of the range should keep the consistency as in the density plot which we use `a-b` without the spaces. We should probably consolidate the code a bit. The related function is `getBinStatsForTooltip` in StudyViewUtils.

Reply:
This is a good suggestion - I will put my logic in there and reuse what's already there if I can. Thank you.